### PR TITLE
chore(deps): Update Eslint

### PR DESCRIPTION
**Pull request by imoutaharik**
And fixing this warning while running `pnpm run build` or `pnpm storybook`
Screenshot 2023-04-28 at 13 03 01

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -173,6 +173,8 @@
         "@react-types/dialog": "3.4.5",
         "@react-types/shared": "3.16.0",
         "@tailwindcss/forms": "^0.5.3",
+        "@typescript-eslint/eslint-plugin": "^5.59.1",
```
Comment (SamuelAlev) [inline package.json:176]: Update the Frontify config instead, it is the reason

**Review by SamuelAlev — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (imoutaharik) [reply package.json:176]: Thanks, Right! I just noticed this Pr: https://github.com/Frontify/fondue/pull/1367/files
There are some linting issue to take care of as well. Taking a look in a bit.

File: src/components/Popper/Popper.tsx [hunk 2]
```diff
@@ -24,6 +24,7 @@ const Reference = ({ children }: { children: ReactElement }) => {
 
 const Content = ({ children }: { children?: ReactNode }) => {
     const { open } = usePopperContext();
+    // eslint-disable-next-line react/jsx-no-useless-fragment
```
Comment (SamuelAlev) [inline src/components/Popper/Popper.tsx:27]: Same

File: scripts/generateIconEnum.ts [hunk 3]
```diff
@@ -7,7 +7,7 @@ const ICON_ENUM_OUTPUT_PATH = './src/foundation/Icon/IconEnum.ts';
 
 (async () => {
     const iconFilePath = await fastGlob('./src/foundation/Icon/Generated/*.tsx', { objectMode: true });
-    const iconNames = iconFilePath.map((path) => path.name.replace(/(^Icon)|(\.tsx$)/g, ''));
+    const iconNames = iconFilePath.map((path) => path.name.replaceAll(/(^Icon)|(\.tsx$)/g, ''));
```
[the remark this answers is not in the document]
Comment (fabriciomendonca) [reply scripts/generateIconEnum.ts:10]: I found the docs here for the rule. Maybe we don't need the global flag anymore?
https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-string-replace-all.md

[the remark this answers is not in the document]
Comment (imoutaharik) [reply scripts/generateIconEnum.ts:10]: Thanks, good catch! I forgot to remove the `global`.

[the remark this answers is not in the document]
Comment (imoutaharik) [reply scripts/generateIconEnum.ts:10]: Actually, _If pattern [is a regex](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/includes), then it must have the global (g) flag set, or a [TypeError](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypeError) is thrown._
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replaceAll

[the remark this answers is not in the document]
Comment (fabriciomendonca) [reply scripts/generateIconEnum.ts:10]: TIL! Didn't know that, always used `replace` for regex and `replaceAll` for strings. Thanks for sharing :D

File: src/components/Tree/utils/removeFragmentsAndEnrichChildren.spec.cy.tsx [hunk 4]
```diff
@@ -85,7 +85,7 @@ describe('removeFragmentsAndEnrichChildren', () => {
     });
 
     it('should handle empty fragments', () => {
```
Comment (fabriciomendonca) [inline src/components/Tree/utils/removeFragmentsAndEnrichChildren.spec.cy.tsx:87]: Maybe changing the test title to `should handle nullable children`?

Comment (imoutaharik) [reply src/components/Tree/utils/removeFragmentsAndEnrichChildren.spec.cy.tsx:87]: Actually it needs to be removed as it's already tested in line 80.

**Review by fabriciomendonca — APPROVED** [review]
It looks good from my side.

File: src/components/Popover/Popover.tsx [hunk 5]
```diff
@@ -16,6 +16,7 @@ const Trigger = ({ children }: { children: ReactElement }) => {
 };
 
 const Content = ({ children }: { children?: ReactNode }) => {
+    // eslint-disable-next-line react/jsx-no-useless-fragment
```
[the remark this answers is not in the document]
Comment (SamuelAlev) [reply src/components/Popover/Popover.tsx]: All good, I digged into a bit more and IMO fine

**Review by SamuelAlev — APPROVED** [review]

**Pull request merged**